### ISSUE TITLE
docs: Issue in page Rolling Restart Procedure

### ISSUE THREAD

**Issue by tnozicka** [scylladb/scylladb#13156]
I would like to report an issue in page https://docs.scylladb.com/branch-5.1/operating-scylla/procedures/config-change/rolling-restart

### Problem
The rolling restart procedure should maintain HA cluster that is accessible using CL_QUORUM during the rolling restart. While it may seems like taking only one node down maintains that invariant, checking that the particular node has become UN has proven not to be enough to make sure that node is a part of the cluster and the other nodes can still see it as DN. That means the next node can be taken down too early and CL_QUORUM queries will fail. Both Scylla Operator and Scylla Cloud use this procedure of checking only UN of the local node, from its point of view, possibly accompanied by sleep or checking CQL port. The issue showing that this procedure wasn't enough is https://github.com/scylladb/scylla-operator/issues/1077, https://github.com/scylladb/scylla-operator/issues/1077#issuecomment-1328791337 discusses the UN issue. (Let's leave the discussed solution on the issue and related PR aside as we are still trying to figure out what to do there.)

###  Suggest a fix
We need to update the procedure to make sure the rolling restart maintains HA cluster, iow. CL_QUORUM keeps working all the time (on the happy path). What the procedure is, I am unsure of. 

Before raft, there are options like asking all the other nodes if they also think this node is UN. With raft that could be asking the leader whether he (aka the majority) thinks this node is UN. For Scylla Operator this also has other implications as we have individual readiness probes that are then aggregated so having to depend on cluster state (with N-1) will either make all ready, or none but I guess we'd have to deal with it anyways. In any case I was wondering what the recommended procedure on VM deployment would be.

**Comment by vladzcloudius** [comment]
I don't think this is an issue and I don't think that a "LOCAL_QUORUM" HA must be a requirement in a general case.

**Comment by mykaul** [comment]
@eliransin - can you assign someone to help the team to figure out the (potential? existing?) race here and how to properly handle it?
